grpc.go

```diff
@@ -434,12 +434,13 @@ func (a *agentGRPC) postExecProcess(ctr *container, proc *process) error {
 // path set by the spec, since we will always ignore it. Indeed, it makes no
 // sense to rely on the namespace path provided by the host since namespaces
 // are different inside the guest.
-func (a *agentGRPC) updateContainerConfigNamespaces(config *configs.Config, ctr *container) {
+func (a *agentGRPC) updateContainerConfigNamespaces(config *configs.Config, ctr *container) error {
 	var ipcNs, utsNs bool
 
 	for idx, ns := range config.Namespaces {
-		// TODO: NEW_PID should be cleared out by the runtime. Add a strict check
-		// for this to return an error if NEWPID is found once the runtime PR lands.
+		if ns.Type == configs.NEWPID {
```

Comment:
I'm slightly in two minds here. We know we are not going to use the PID namespace in the agent don't we, so is this really an error - does it actually affect any functionality? I wonder if:
- We could make it a warning, not an error
- We could move the check to the runtime?, as adding it here grows the agent (sure, only a fraction), inside the container - and really we'd like to try and keep our footprint overhead as small as we can.

So, nothing wrong with the patch, and lgtm - I just thought I'd ask if it's wholly necessary and in the best place to do the check?

Reply:
Ah - your ack hasn't applied @grahamwhaley as it's not at start of line - was that intended?

Reply:
It was :-) - I thought we'd discuss a touch first.

Reply:
I would prefer to do the checking at the beginning of the grpc request handler so that we do not have to revert anything if the sanity check fails.

Reply:
@bergwolf Fair enough, I'll add this check early on.